components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.cpp

```diff
@@ -461,5 +465,44 @@ void SHOCMacrophysics::finalize_impl()
   // Do nothing
 }
 // =========================================================================================
+void SHOCMacrophysics::check_flux_state_consistency(const double dt)
+{
+  using PC = scream::physics::Constants<Real>;
+  const Real gravit = PC::gravit;
+
+  const auto& pseudo_density = get_field_in ("pseudo_density").get_view<const Spack**>();
+  const auto& surf_evap      = get_field_out("surf_evap").get_view<Real*>();
+  const auto& qv             = get_field_out("qv").get_view<Spack**>();
+
+  const auto nlevs      = m_num_levs;
+  const auto nlev_packs = ekat::npack<Spack>(nlevs);
+  const auto policy     = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(m_num_cols, nlev_packs);
+  Kokkos::parallel_for("check_flux_state_consistency",
+                       policy,
+                       KOKKOS_LAMBDA (const KT::MemberType& team) {
+    const auto i = team.league_rank();
+
+    const auto& pseudo_density_i = ekat::subview(pseudo_density, i);
+    const auto& qv_i             = ekat::subview(qv, i);
 
+    if (surf_evap(i) < 0) {
```

Comment:
It seems as if we are applying adjustment anytime the surface latent flux is less than zero (i.e. anytime cc < 0)?  Is that true @oksanaguba?  If so then I do not agree with this as cc < 0 (surface latent flux less than zero) is something that can and does happen quite often in nature and the model and SCREAM can handle this gracefully in 99.999999% of instances... However there are a few instances (primarily during spin-up) when the negative surface latent heat flux can exhaust the moisture in the lowest model level.  It is only in these outlier cases where the fixer/adjust should be applied.  Therefore I propose replacing line 

```
if (cc < 0.0_r8) then
```

with a trigger something in the spirit in qneg4.F90

```
 excess(i) = qflx(i,1) - (qmin(1) - qbot(i,1))/(ztodt*gravit*srfrpdel(i)) 
  
 if (excess(i) < 0._r8) then 
```

Where it checks to see if the bottom layer vapor will be exhausted or not.  If it is, then carry on with Oksana's fixer.  Else, if we do this whenever cflx < 0 if we will be overly applying this adjustment and I anticipate that simulation quality will be deteriorated.